hil/ext/switches/brocade.py

```diff
@@ -265,6 +271,89 @@ def _disable_native_tag(self, interface):
         url = self._construct_url(interface, suffix='trunk/tag/native-vlan')
         self._make_request('DELETE', url, acceptable_error_codes=(404,))
 
+    def save_running_config(self):
+        url = self.hostname + '/rest/' + 'operational-state/bna-config-cmd'
+        body = '<bna-config-cmd><src>running-config</src><dest>startup-config \
+                </dest></bna-config-cmd>'
+        response = self._make_request('POST', url, data=body)
+
+        # check if it succesfully completed and log it
+        tag = '{urn:brocade.com:mgmt:brocade-ras}session-id'
+        root = etree.fromstring(response.text)
+        session_id = root.find(tag).text
+        if self._check_bna_config_status(session_id):
+            logger.debug('Saved running config to startup config')
+        else:
+            logger.debug('Configuration not saved in reasonable time')
+
+    def get_config(self, config_type):
+        # This is the only method that uses pexpect to connect to the switch
+        # because the REST API does not return the startup config file.
+        # This method is only used by the test suite, the rest of the switch
+        # functionality does not care about this method.
+
+        # this is how we get the running config file using the API
+        # I couldn't find a way to get the startup config file similarly.
+        # url = self.hostname + '/rest/config/' + config_type
+        # response = self._make_request('GET', url)
+
+        from hil.ext.switches._console import login, get_prompts
+
+        # is this safe to assume that there will be http in front of the
+        # hostname? also create a deep copy rather than modifying the original
+        # object
+        original_hostname = self.hostname
+        self.hostname = self.hostname.replace('http://', '')
+        console = login(self)
+        self.hostname = original_hostname
+
+        console.sendline('')
+        prompts = get_prompts(console)
```

Comment:
why am I getting the prompts when I don't even use them

Reply:
maybe you don't want to save `get_prompts(console)` as a variable called `prompts` here. Since you are not using it later on in the code.

Reply:
yeah, ill remove that.